analyzer/codechecker_analyzer/cmd/analyze.py

```diff
@@ -757,6 +766,33 @@ def __get_result_source_files(metadata):
     return result_src_files
 
 
+def __have_new_report(plist_timestamps, output_path):
+    """
+    This is a lightweight implementation of checking whether new reports are
+    found during analysis. The function checks if a new plist file contains
+    "check_name" string. There is no plist parsing or "CodeChecker parse" run,
+    but we say that a report was emitted if any checker writes anything to a
+    plist file. "New plist file" means that it is either created during the
+    analysis or its timestamp is greater than before analysis.
+
+    plist_timestamps -- A dict storing file names and timestamps before
+                        analysis.
+    output_path -- Path of the output directory containing plist files.
+    """
+    for f in os.listdir(output_path):
+        if not f.endswith('.plist'):
+            continue
+
+        abs_f = os.path.join(output_path, f)
+        if f not in plist_timestamps or \
+                os.path.getmtime(abs_f) > plist_timestamps[f]:
+            with open(abs_f, encoding='utf-8', errors='ignore') as plist:
+                if 'check_name' in plist.read():
```

Comment:
Maybe the `'check_name'` can be a parameter of this function or we can create a separate variable inside this function so we can see more what we check here.

Reply:
I intentionally hard-coded this .plist tag in this function, because purpose of this function is to determine whether any report was found by CodeChecker. The fact that this is determined by `check_name` is just an implementation detail, so I don't think it should be the parameter of this function.
I don't understand how this can be a separate variable, could you please describe this suggestion a little more?
But I wrote this down in some documentation comment why this is checked.